### PR TITLE
fix: close the 14 confirmed unhinted-adversarial-review findings, bump 1.0.0

### PR DESCRIPTION
**Pull request by anaclumos**
Closes the 14 confirmed findings from the unhinted whole-repo adversarial review (10 independent lenses, every finding adversarially verified before acceptance; 6 candidates were refuted). Bumps to 1.0.0 for the release that follows the merge. The literal Codex-runtime review is owed after Jul 25 (its account hit the usage limit); this review ran as the Claude-side substitute the owner picked.

## HIGH

- **Positional prompt replay (supervisor)**: `claude "do the thing" --model opus` persisted the prompt like a flag and replayed it verbatim on every depleted-pool respawn and any later bare `claude --resume <id>` - re-submitting the original instruction into an already-progressed session. `stripPositionals` (flag-classification aware: value-taking, variadic, optional-value) keeps flags only for `sessions/` files and respawn args; the first launch alone carries the prompt. Pinned by a new behavioral respawn-loop test.
- **`config unset` could brick loadConfig**: it wrote without the merged-whole validation `config set` performs, so removing one override could persist a file whose merged result fails the projectionMargin-vs-thresholds refine - every later `loadConfig()` throws, disabling status/switch/hooks/statusline until hand-repaired. The same gate now runs on unset.
- **Stop hook had zero behavioral coverage**: new `test/stophook.test.ts` drives the real `__stop-hook` subprocess (hermetic single-account pool, depleted-wait resolving to the current seat - no credentials, no network) and pins: marker written ONLY for a depleted wait, marker FILE keyed by the pinned env sid with the stdin sid in the payload, non-UUID stdin falling back to the pinned sid, and no marker unsupervised.

## MEDIUM

- **Trigger/screening disagreement on null-reset windows** (`decide.ts liveUsed`): the trigger side kept reading a stale over-bar row (reset clock unparseable, anti-storm carrying it forward) as live forever, while the picker's `blockedUntil` self-bounds at sampledAt + window duration - the two halves of one decision disagreed, forcing hard-path swaps or waitUntil=now respawn churn off a healthy account. `liveUsed` now applies the same self-bound (mirroring the codex analog).
- **Drain vs Slack-delivery failure** (`serve.ts` presumedKilled): a turn whose child reached a successful result but whose final Slack append failed (textLost) was presumed killed during a drain, retaining the resume marker - the next generation re-ran completed work. `TurnOutcome.resultReceived` distinguishes the classes; daemon test added.
- **`rm --codex`**: codex accounts were unremovable while the uninstall message pointed at `xx rm` for every parked credential. New `cmdCodexRm` under the codex flock, refusing the live account (identity decoded from the blob, fail-fast on an unreadable one) and present (running) accounts; help text + command tables updated in EN and all 29 locales.
- **Respawn loop behavioral test**: marker consume -> SIGTERM -> `--resume <marker sid>` relaunch, plus the invalid-marker drop (no kill, no crash, marker deleted).
- **Banned `as`-casts** in `probe-hang.test.ts` / `swap-concurrency.ts` replaced with the sanctioned structural narrowing / a correct env type.
- **DESIGN.md**: the false absolute "this is the only time CLAUDE_CONFIG_DIR is ever used" corrected (auth and parked-account sampling use the same isolation; ambient values are refused).

## LOW

- **Serve queue overflow**: chat 4.34.0 defaults maxQueueSize 10 with a LOG-LESS drop-oldest trim, silently eating the oldest messages of a >10 burst behind a long turn; now 100 (the TTL stays the real bound).
- **Duplicate labels**: `rename` (both pools) refuses a label another account already uses - selectors resolve first-match, so a dupe misdirected `rm`/`switch`/`auth`.
- **Codex switch empty selector**: `""` matched the first account via `startsWith("")`; now truthiness-guarded like the claude switch.
- **README**: the plain `tokenmaxxing switch` row was missing from the commands table.

## Verification

- `bun run typecheck` clean
- `bun test`: 409 pass (7 new behavioral tests) / 1 skip / 0 fail
- `bun test/e2e/swap-concurrency.ts`: ALL PASS (decision path touched)
- banned-character grep over the diff: clean

---

> [!NOTE]
> **Medium Risk**
> Touches supervisor respawn args, swap decision logic, config persistence, credential deletion, and Slack drain/resume markers—critical paths, but changes are defensive fixes with new behavioral tests.
> 
> **Overview**
> **Release 1.0.0** bundles fixes from the adversarial review across session safety, switching, config, Codex pool management, and Slack serve.
> 
> The **supervisor** now persists and replays **flags only** via `stripPositionals`, so one-shot positional prompts are not re-submitted on depleted-pool respawn or bare `--resume`; legacy `sessions/` files are sanitized on restore. **Stop hook** and **respawn-loop** behavior is covered by new subprocess tests.
> 
> **`config unset`** runs the same merged-config validation as **`set`**, avoiding a bad override removal that could make every `loadConfig()` throw. **`decide.ts`** aligns `liveUsed` with the picker for windows with a missing reset time, so trigger and screening no longer disagree.
> 
> **Slack serve** sets `maxQueueSize: 100`, adds **`resultReceived`** on turn outcomes, and during drain keeps resume markers only when the child did not finish successfully—so Slack delivery failures do not re-run completed work.
> 
> **Codex**: new **`tokenmaxxing rm --codex`**, `deleteParkedCodexAuth`, bare **`credFile`** names in Zod, empty-selector fix on **`switch --codex`**, and **case-insensitive unique labels** on rename for both pools. Docs/README gain **`switch`**, **`rm [--codex]`**, and clarified **`CLAUDE_CONFIG_DIR`** usage in DESIGN.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 932fb419685953699e95615a5133c74b2b9ead04. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

---
## Summary by cubic
Fixes 14 confirmed issues from the whole-repo adversarial review to harden session safety, account switching, and Slack relay reliability. Bumps `tokenmaxxing` to 1.0.0.

- Bug Fixes
  - Supervisor: prevent positional prompt replay on respawn/resume; persist flags only and honor `--` end-of-options; sanitize old session files on restore; adds a respawn-loop test.
  - `config unset`: validate the merged config like `set` to avoid bricking `loadConfig()`.
  - Stop hook: add behavioral tests; write a respawn marker only on depleted waits; correct session-id keying and fallback; none when unsupervised.
  - Decision logic: align null-reset window handling in `decide.ts` so trigger and screening agree; prevents churny swaps/respawns.
  - Slack drain: if the child produced a successful result, clear the marker even when Slack delivery later fails, so completed work isn’t re-run.
  - Codex removal: add `tokenmaxxing rm --codex`; refuse live or running accounts; actually delete the parked blob via a single-owner path; validate `credFile` as a bare file name (no path) to prevent traversal; help/docs updated across EN and 29 locales, README shows `rm [--codex]`.
  - Slack bridge queue: set `maxQueueSize: 100` to avoid silent drop-oldest; TTL unchanged.
  - Pool hygiene: enforce unique labels in both pools (case-insensitive); treat empty `tokenmaxxing switch --codex ""` as no selector.
  - Docs: clarify `CLAUDE_CONFIG_DIR` usage in DESIGN; add missing `tokenmaxxing switch` row in README.

<sup>Written for commit 932fb419685953699e95615a5133c74b2b9ead04. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/anaclumos/tokenmaxxing/pull/37?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/cli/rename.ts [hunk 1]
```diff
@@ -39,6 +39,14 @@ async function renameCodexAccount(input: { selector: string; newLabel: string })
       console.error(c.red(`no codex account matches "${input.selector}"`));
       return 1;
     }
+    // labels resolve selectors first-match: a duplicate would make the other
+    // account unreachable by label and misdirect destructive commands like
+    // `rm` onto the wrong one (adversarial-review catch)
+    const taken = index.accounts.find((x) => x.accountId !== account.accountId && x.label === input.newLabel);
+    if (taken) {
+      console.error(c.red(`label "${input.newLabel}" is already used by ${taken.accountId.slice(0, 8)} - labels must be unique within the pool`));
+      return 1;
```
[the remark this answers is not in the document]
Comment (anaclumos) [reply src/cli/rename.ts:50]: Agreed and fixed in 55d4c71: both guards now compare label.toLowerCase(), matching the case-insensitive selector resolution in findAccount/findCodexAccount.

File: src/entries/supervisor.ts [hunk 2]
```diff
@@ -274,7 +297,10 @@ export async function runSupervisor(argv: string[]): Promise<number> {
     const persisted = loadSessionFlags(sid);
     if (persisted) base = persisted;
   }
-  saveSessionFlags(sid, base, process.cwd());
+  // The FIRST launch keeps a positional prompt (the user just typed it);
+  // everything persisted or respawned carries flags only.
+  const persistable = stripPositionals(base);
+  saveSessionFlags(sid, persistable, process.cwd());
```
[the remark this answers is not in the document]
Comment (anaclumos) [reply src/entries/supervisor.ts:323]: Agreed and fixed in 55d4c71: the restore path runs stripPositionals on the loaded flags (enforcing the flags-only contract at the trust boundary; idempotent on well-formed files), so a pre-fix stored positional never rides a bare --resume. Regression test added. User-typed positionals on the command line itself still submit once, as intended.

File: src/cli/codexrm.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      return 1;
+    }
+    if (presentCodexAccountIds().has(account.accountId)) {
+      console.error(c.red(`${account.label} is running in a live codex session - close that session before removing it.`));
+      return 1;
+    }
+    // Parked codex blobs are plain 0600 files; hard delete on purpose (the
+    // credential-dir cleanup exception - trashing would move a credential
+    // into the Trash folder).
+    rmSync(join(codexPaths.credsDir, account.credFile), { force: true });
```
[the remark this answers is not in the document]
Comment (anaclumos) [reply src/cli/codexrm.ts]: Agreed and fixed in 6bda408: deleteParkedCodexAuth in codexauth.ts owns the parked-path shape (the .json suffix) and rm --codex uses it - the hand-built extensionless path is gone.

File: src/entries/supervisor.ts [hunk 4]
```diff
@@ -155,6 +155,29 @@ export function stripSessionFlags(argv: string[]): string[] {
...
+ *  flag and its consumed value(s). A positional is a submit-once user turn:
+ *  persisting or replaying it on a respawn / later `--resume` re-injects the
+ *  original instruction into an already-progressed session (adversarial-review
+ *  HIGH catch) - only real flags like --model belong in sessions/ files and
+ *  respawn args. */
+export function stripPositionals(argv: string[]): string[] {
+  const out: string[] = [];
+  for (let i = 0; i < argv.length; i++) {
+    const a = argv[i]!;
+    if (!a.startsWith("-")) continue;
```
[the remark this answers is not in the document]
Comment (anaclumos) [reply src/entries/supervisor.ts:183]: Agreed and fixed in 6bda408: stripPositionals breaks on the -- delimiter, so everything after it (a prompt deliberately starting with a dash) stays positional and is never persisted or replayed.

File: README.md [hunk 5]
```diff
@@ -39,6 +39,7 @@ claude                  # use claude as always
 | `tokenmaxxing add` | register an additional account (isolated login, harvested into the pool) |
 | `tokenmaxxing add --codex` | register an additional codex account (isolated login) |
 | `tokenmaxxing auth [sel \| --all]` | reauthenticate a pooled account in place: bare lists the pool (emails shown) and asks which; a selector targets one account and tells you the email to sign in with; `--all` walks every needs-reauth account one by one |
+| `tokenmaxxing switch [sel]` | switch the claude pool: bare picks the best account greedily (no-op when the current one wins), a selector targets one |
```
[the remark this answers is not in the document]
Comment (anaclumos) [reply README.md:42]: Agreed and fixed in 6bda408: the README row now reads rename [--codex] <sel> <label> / rm [--codex] <sel>.

File: src/entries/supervisor.ts [hunk 6]
```diff
@@ -155,6 +155,29 @@ export function stripSessionFlags(argv: string[]): string[] {
...
+    out.push(a);
+    if (VALUE_TAKING_ROOT_FLAGS.has(a)) {
+      if (i + 1 < argv.length) out.push(argv[++i]!);
+    } else if (VARIADIC_ROOT_FLAGS.has(a)) {
+      while (i + 1 < argv.length && !argv[i + 1]!.startsWith("-")) out.push(argv[++i]!);
+    } else if (OPTIONAL_VALUE_ROOT_FLAGS.has(a)) {
+      if (i + 1 < argv.length && !argv[i + 1]!.startsWith("-")) out.push(argv[++i]!);
+    }
+  }
+  return out;
```
[the remark this answers is not in the document]
Comment (anaclumos) [reply src/entries/supervisor.ts:193]: Agreed and fixed in 6bda408 (same root cause as the cubic thread): stripPositionals stops at --, dropping the delimiter and everything after it.

File: src/lib/codexauth.ts [hunk 7]
```diff
@@ -65,6 +65,13 @@ export function writeParkedCodexAuth(input: { credFile: string; auth: CodexAuthJ
   writeFileAtomic(parkedPath(input), JSON.stringify(CodexAuthJsonSchema.parse(input.auth), null, 2), 0o600);
 }
 
+/** `rm --codex` uses this so the path shape (.json suffix) has one owner:
+ *  a hand-built path without the suffix silently missed the real file under
+ *  rmSync force (PR #37 review catch). */
+export function deleteParkedCodexAuth(input: { credFile: string }): void {
+  rmSync(parkedPath(input), { force: true });
```
[the remark this answers is not in the document]
Comment (anaclumos) [reply src/lib/codexauth.ts:72]: Agreed and fixed in 932fb41, one level up from the suggestion: `credFile` is now a bare-file-name schema (`BareFileNameSchema` in types.ts) rather than a check at the delete site, so a corrupted `codex-accounts.json` fails the load loudly - matching the codex loaders throw-on-unparsable contract - and the read/write paths are covered along with the delete. Pinned by a new contract test in test/pure.test.ts.

File: src/entries/supervisor.ts [hunk 8]
```diff
@@ -155,6 +155,33 @@ export function stripSessionFlags(argv: string[]): string[] {
...
```
[the remark this answers is not in the document]
Comment (anaclumos) [reply src/entries/supervisor.ts:193]: WONTFIX, with the tradeoff now documented in-file (932fb41).

The mechanism is real: a value-taking root flag claude adds before these sets are updated has its value read as a positional and dropped. But the alternative default - treating a bare token after an *unrecognized* flag as that flag's value - moves the same staleness into a silent failure: a newly added BOOLEAN flag sitting before the prompt would make the prompt look like a value and replay a submit-once user turn into an already-progressed session, which is the exact harm stripPositionals exists to prevent.

Neither default is safe under a stale set; the ambiguity is irreducible without claude's own option table. Current behavior fails loudly (claude errors on the missing argument and the user relaunches), the alternative fails silently. Keeping the loud one and documenting why, per the repo rule on flagged tradeoffs.
